jenkinsfile: fix paths for all steps

markoburcul opened the pull request:
Since nix shell function from jenkins librarz uses WORKSPACE env variable to find shell.nix we need to override it for steps using nix shell. For all of the steps I'm using dir directive to change cwd to the apps/connector.

Referenced issue: https://github.com/status-im/status-web/issues/590

markoburcul committed:
jenkinsfile: fix paths for all steps

Since nix shell function from jenkins librarz uses WORKSPACE env
variable to find shell.nix we need to override it for steps using nix
shell. For all of the steps I'm using dir directive to change cwd to the
apps/connector.

Referenced issue: https://github.com/status-im/status-web/issues/590

Signed-off-by: markoburcul <marko@status.im>

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | ff85810e | [#1](https://ci.status.im/job/status-web/job/prs/job/linux/job/PR-597/1/) | 2024-10-10 11:28:56 | ~36 sec | `chrome` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Extension-241010-112818-ff8581-pr597-chrome.zip) |

jakubgs commented on apps/connector/Jenkinsfile:
```diff
@@ -28,40 +28,60 @@ pipeline {
-        nix.shell('yarn install --frozen-lockfile', pure: false)
-      } }
+      steps {
+          dir("${env.WORKSPACE}/apps/connector") {
+            script {
+              withEnv(["WORKSPACE=${env.WORKSPACE}/apps/connector"]) { // Override WORKSPACE
```
That's way too overcomplicated. Wouldn't it be better to just add support for an `entrypoint` argument for `nix.shell()` and other commands in `status-jenkins-lib`?

markoburcul replied:
I agree with that, will fix it right away

jakubgs reviewed (approved):
I don't want to block you from fixing stuff, so I'm approving, but I would just fix this in `status-jenkins-lib` if I was you.

markoburcul committed:
jenkinsfile: use newest status jenkins lib

Use newest jenkins lib tag which adds the entryPoint as an argument to the nix shell function.

Referenced issue: https://github.com/status-im/status-web/issues/590

Signed-off-by: markoburcul <marko@status.im>
